packages/react-native-reanimated/Common/cpp/reanimated/CSS/common/transforms/TransformMatrix2D.cpp

```diff
@@ -1,15 +1,149 @@
 #include <reanimated/CSS/common/transforms/TransformMatrix2D.h>
 
+#include <cmath>
+
 namespace reanimated::css {
 
+TransformMatrix2D::TransformMatrix2D(jsi::Runtime &rt, const jsi::Value &value)
+    : TransformMatrixBase<TransformMatrix2D, MATRIX_2D_DIMENSION>() {
+  const auto &array = value.asObject(rt).asArray(rt);
+  const auto size = array.size(rt);
+
+  if (size == SIZE) {
+    // Direct 3x3 matrix
+    for (size_t i = 0; i < SIZE; ++i) {
+      matrix_[i] = array.getValueAtIndex(rt, i).asNumber();
+    }
+  } else {
+    // Convert 4x4 matrix to 3x3
+    // (we should call canConstruct first to ensure that the matrix is
+    // convertible to 3x3)
+    //
+    // [m[0], m[1], 0, m[2]]     [m[0], m[1], m[2]]
+    // [m[3], m[4], 0, m[5]]  -> [m[3], m[4], m[5]]
+    // [  0,   0,  1,   0]       [m[6], m[7], m[8]]
+    // [m[6], m[7], 0, m[8]]
+    matrix_[0] = array.getValueAtIndex(rt, 0).asNumber();
```

Comment:
Let's add checks for the size of the array, or at least for size 0, to ensure we're receiving what we expect from JS.